### PR TITLE
improvement(longevity-all-compressions): Adding new longevity

### PR DESCRIPTION
**Pull request by roydahan**
Addig a new longevity with stress commands that uses all types
of compressions available.

## PR pre-checks (self review)

- [ ] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [ ] I didn't leave commented-out/debugging code
- [ ] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Review by juliayakovlev — APPROVED** [review]

File: jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!groovy
+
+// trick from https://github.com/jenkinsci/workflow-cps-global-lib-plugin/pull/43
+def lib = library identifier: 'sct@snapshot', retriever: legacySCM(scm)
+
+longevityPipeline(
+    backend: 'aws',
+    aws_region: 'eu-west-1',
+    test_name: 'longevity_test.LongevityTest.test_custom_time',
+    test_config: 'longevity-all-compressions-3days.yaml',
```
Comment (bentsi) [inline jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile:10]: need to mention the full path
`test-cases/longevity/longevity-all-compressions-3days.yaml`

File: jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+    timeout: [time: 4600, unit: 'MINUTES']
```
Comment (bentsi) [inline jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile:12]: no need to mention timeout, it is calculated from `test_duration`

File: test-cases/longevity/longevity-all-compressions-3days.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+gce_instance_type_loader: 'c2-standard-16'
+scylla_repo_loader: 'https://s3.amazonaws.com/downloads.scylladb.com/rpm/centos/scylla-4.3.repo'
+
+
+nemesis_class_name: 'ChaosMonkey'
+nemesis_interval: 30
+nemesis_filter_seeds: false
+seeds_num: 2
+
+user_prefix: 'longevity-tls-1tb-7d'
```
Comment (bentsi) [inline test-cases/longevity/longevity-all-compressions-3days.yaml:39]: Should be `longevity-all-compressions-3days`

File: jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile (hunk 2 above)
Comment (bentsi) [reply jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile:12]: better to set test duration to 4600

**Review by bentsi — CHANGES_REQUESTED** [review]
please address comments

File: jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile (hunk 2 above)
Comment (roydahan) [reply jenkins-pipelines/longevity-all-compressions-3days.jenkinsfile]: Removed.

**Comment by roydahan** [comment]
Not tested yet.

File: test-cases/longevity/longevity-all-compressions-3days.yaml [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+n_loaders: 4
+n_monitor_nodes: 1
+
+instance_type_db: 'i3.4xlarge'
+gce_instance_type_db: 'n1-highmem-16'
+gce_n_local_ssd_disk_db: 6
+
+instance_type_loader: 'c5.4xlarge'
+gce_instance_type_loader: 'c2-standard-16'
+scylla_repo_loader: 'https://s3.amazonaws.com/downloads.scylladb.com/rpm/centos/scylla-4.3.repo'
```
Comment (amoskong) [inline test-cases/longevity/longevity-all-compressions-3days.yaml:31]: Do we want to set it to 4.4 repo?

File: test-cases/longevity/longevity-all-compressions-3days.yaml [hunk 5]
```diff
@@ -0,0 +1,50 @@
+test_duration: 4320
+prepare_write_cmd: [ "cassandra-stress write cl=QUORUM n=50000000 -schema 'replication(factor=3) compression=LZ4Compressor'     -port jmx=6868 -mode cql3 native compression=lz4     -rate threads=100 -pop seq=1..50000000 -log interval=5",
```
Comment (amoskong) [inline test-cases/longevity/longevity-all-compressions-3days.yaml:2]: What's the dataset size?  why not limit the column size by `-col 'size=FIXED(1024) n=FIXED(1)'`?

File: test-cases/longevity/longevity-all-compressions-3days.yaml (hunk 4 above)
Comment (roydahan) [reply test-cases/longevity/longevity-all-compressions-3days.yaml:31]: what is it anyway?

File: test-cases/longevity/longevity-all-compressions-3days.yaml (hunk 5 above)
Comment (roydahan) [reply test-cases/longevity/longevity-all-compressions-3days.yaml:2]: these are the original stress commnads taken out from the 1tb longevity.
They are failing, and we need to test why.

**Review by juliayakovlev — APPROVED** [review]

**Comment by enaydanov** [comment]
@bentsi can it be merged? There are no conflicts and mocked_tests failed because this branch created earlier than these tests was introduced.

File: test-cases/longevity/longevity-all-compressions-3days.yaml (hunk 4 above)
Comment (bentsi) [reply test-cases/longevity/longevity-all-compressions-3days.yaml:31]: we use it to install c-s on non-AWS tests and it is defined in `test_default.yaml`, so you should remove it

**Comment by bentsi** [comment]
@roydahan there is a small comment to address, and we can merge this.

**Comment by fruch** [comment]
@roydahan this PR is still needed ?

**Comment by roydahan** [comment]
It’s needed, but someone need to actually test it.
All these commands used to be in the 1TB longevity, but we split them out.
I can try assigning it to someone who relatively new to this for testing.

File: test-cases/longevity/longevity-all-compressions-3days.yaml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+stress_cmd: [ "cassandra-stress mixed cl=QUORUM duration=4200m -schema 'replication(factor=3) compression=LZ4Compressor'       -port jmx=6868 -mode cql3 native compression=lz4     -rate threads=50 -pop seq=1..50000000 -log interval=5",
+              "cassandra-stress mixed cl=QUORUM duration=4200m -schema 'replication(factor=3) compression=SnappyCompressor'    -port jmx=6868 -mode cql3 native compression=snappy  -rate threads=50 -pop seq=1..50000000 -log interval=5",
+              "cassandra-stress mixed cl=QUORUM duration=4200m -schema 'replication(factor=3) compression=DeflateCompressor'   -port jmx=6868 -mode cql3 native compression=none    -rate threads=50 -pop seq=1..50000000 -log interval=5",
+              "cassandra-stress mixed cl=QUORUM duration=4200m -schema 'replication(factor=3) compression=ZstdCompressor'      -port jmx=6868 -mode cql3 native compression=none    -rate threads=50 -pop seq=1..50000000 -log interval=5" ]
+
+stress_read_cmd: [ "cassandra-stress read cl=QUORUM duration=4200m -schema 'replication(factor=3) compression=LZ4Compressor'     -port jmx=6868 -mode cql3 native compression=lz4    -rate threads=20 -pop seq=1..50000000 -log interval=5",
```
Comment (fruch) [inline test-cases/longevity/longevity-all-compressions-3days.yaml:12]: remove `-port jmx=6868` from all commands

**Comment by fruch** [comment]
@roydahan I see you archived the task in trello, any point of leaving this PR open ?

**Pull request closed without merging**